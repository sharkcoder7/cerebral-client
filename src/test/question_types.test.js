import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import './setup_tests'
import YesNoDetails from '../components/question_types/yes_no_details'
import Date from '../components/question_types/date'

describe('Yes No Component testing', () => {
  let component = null;
  let check_date = null;
  const test_action = data => {
    check_date = data;
  }

  it('renders correctly', () => {
    component = shallow(<Date submit_action= {test_action} />); 
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('componet check', () => {
    expect(component.find('input').exists()).toBe(true);
  });
  
  it('simulates input change', () => {
    const mockedEvent = {
      target: {
        value: '11/03/87'
      }
    };
    component.find('input').first.simulate('onChange', mockedEvent); 
    expect(component.state().birth_date).toBe('11/03/87');
    });
  
  /*
  it('event check', () => {
    const mockedEvent = {
      preventDefault: () => null 
    }; 
    component.find('button').simulate('onClick', mockedEvent);
  })
  */
});
