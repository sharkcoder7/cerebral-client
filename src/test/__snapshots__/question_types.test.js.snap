// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Yes No Component testing componet check 1`] = `ShallowWrapper {}`;

exports[`Yes No Component testing matches snapshot 1`] = `ShallowWrapper {}`;
